app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    return puts 'No books found' if @books.empty?
+
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    return puts 'No people found' if @people.empty?
+
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    if person_type != '1' && person_type != '2'
+      puts 'Invalid option'
+      return
+    end
+
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    person = if person_type == '1'
+               print 'Has parent permission? [Y/N]: '
+               parent_permission = gets.chomp
+               parent_permission = parent_permission.downcase == 'y'
+
+               Student.new(age, nil, name, parent_permission: parent_permission)
+             else
+               print 'Specialization: '
+               specialization = gets.chomp
+
+               Teacher.new(age, specialization, name)
+             end
+
+    @people << person
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] You did a great job working on this milestone! 🚀  However, I'm having problems trying to create a person, these are the errors I'm getting:

Creating a student:
![image](https://user-images.githubusercontent.com/77069011/232166326-03ee923c-ae04-444a-8b8e-967c5aa582fb.png)

Creating a teacher:
![image](https://user-images.githubusercontent.com/77069011/232166415-3f4d7911-6aa2-4997-99bf-9b17a3adcfc4.png)

Kindly always double-check that your code is running before submitting it. Please consider fixing this! 🤓 🔧

💡 _Hint: Look at your `super()` arguments in both, your `teacher` and `student` classes, and compare them with the params expected in your `person` class, you defined `parent_permission` as a named parameter but you're passing it as a regular one! 🤓_